ballerina/utils.bal

```diff
@@ -244,6 +244,16 @@ isolated function populateCreateObjectHeaders(map<string> requestHeaders, Object
     }
 }
 
+# Function to populate createObject optional user-defined metadata headers.
+#
+# + requestHeaders - Request headers map.
+# + userMetadataHeaders - Map containing user-defined metadata.
+isolated function populateUserMetadataHeaders(map<string> requestHeaders, map<string> userMetadataHeaders){
+    foreach string metadataKey in userMetadataHeaders.keys() {
+        requestHeaders["x-amz-meta-" + metadataKey.toLowerAscii()] = <string>userMetadataHeaders[metadataKey];
```

Comment:
We don't need casting in here, do we?
```suggestion
        requestHeaders["x-amz-meta-" + metadataKey.toLowerAscii()] = userMetadataHeaders[metadataKey];
```

Reply:
Yes, if we don't type cast, there is an error: `incompatible types: expected 'string', found 'string?'`

Reply:
Even in such scenarios its not recommended to use type cast to overcome the error, because you may end up with runtime errors as nil values cannot be cast to strings. Therefore I'd rather go for an alternative like:

```ballerina
requestHeaders["x-amz-meta-" + metadataKey.toLowerAscii()] = userMetadataHeaders[metadataKey] ?: "";
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-aws.s3/commit/dce940d433f4f7e121b4eaf91ea9b3a281dd9aff